src/evo/deterministicmns.cpp

```diff
@@ -175,13 +177,35 @@ static bool CompareByLastPaid(const CDeterministicMN* _a, const CDeterministicMN
     return CompareByLastPaid(*_a, *_b);
 }
 
-CDeterministicMNCPtr CDeterministicMNList::GetMNPayee() const
+CDeterministicMNCPtr CDeterministicMNList::GetMNPayee(const CBlockIndex* pIndex) const
 {
     if (mnMap.size() == 0) {
         return nullptr;
     }
 
-    CDeterministicMNCPtr best;
+    bool isv19Active = llmq::utils::IsV19Active(pIndex);
+    // Starting from v19 and until v20 (Platform release), HPMN will be rewarded 4 blocks in a row
+    // TODO: Skip this code once v20 is active
+    CDeterministicMNCPtr best = nullptr;
+    if (isv19Active) {
+        ForEachMNShared(true, [&](const CDeterministicMNCPtr& dmn) {
+            if (dmn->pdmnState->nLastPaidHeight == nHeight) {
+                // We found the last MN Payee.
+                // If the last payee is a HPMN, we need to check its consecutive payments and pay him again if needed
+                if (dmn->nType == MnType::HighPerformance.index && dmn->pdmnState->nConsecutivePayments < MnType::HighPerformance.voting_weight) {
+                    best = dmn;
+                }
+            }
+            return;
+        });
+
+        if (best)
```

Comment:
if it is single line condition IMO better to write `if (best) return best;` in line line rather than in 2 lines.
Otherwise add {}.

Because it is easy to make a mistake if edit code and remove only "if(best)" or remove only "return best" - the semantic may be changed significantly, but syntax still valid.